connectors/sources/graphql.py

```diff
@@ -0,0 +1,333 @@
+        self.graphql_client.set_logger(self._logger)
+
+    @classmethod
+    def get_default_configuration(cls):
+        """Get the default configuration for GraphQL.
+
+        Returns:
+            dict: Default configuration.
+        """
+        return {
+            "http_endpoint": {
+                "label": "HTTP URL & Endpoint",
+                "order": 1,
+                "type": "str",
+            },
+            "http_method": {
+                "display": "dropdown",
+                "label": "GET/POST",
+                "options": [
+                    {"label": "GET", "value": "get"},
+                    {"label": "POST", "value": "post"},
+                ],
+                "order": 2,
+                "type": "str",
+                "value": "POST",
+            },
+            "authentication_method": {
+                "display": "dropdown",
+                "label": "Authentication Method",
+                "options": [
+                    {"label": "No Auth", "value": "none"},
+                    {"label": "Basic Auth", "value": "basic"},
+                    {"label": "Bearer Token", "value": "bearer"},
+                ],
+                "order": 3,
+                "type": "str",
+                "value": "none",
+            },
+            "username": {
+                "depends_on": [{"field": "authentication_method", "value": "basic"}],
+                "label": "Username",
+                "order": 4,
+                "type": "str",
+            },
+            "password": {
+                "depends_on": [{"field": "authentication_method", "value": "basic"}],
+                "label": "Password",
+                "order": 5,
+                "sensitive": True,
+                "type": "str",
+            },
+            "token": {
+                "depends_on": [{"field": "authentication_method", "value": "bearer"}],
+                "label": "Bearer Token",
+                "order": 6,
+                "sensitive": True,
+                "type": "str",
+            },
+            "graphql_query": {
+                "display": "textarea",
+                "label": "GraphQL Body",
+                "order": 7,
+                "type": "str",
+            },
+            "graphql_variables": {
+                "depends_on": [{"field": "http_method", "value": "post"}],
+                "display": "textarea",
+                "label": "Graphql Variables",
+                "order": 8,
+                "type": "str",
+                "required": False,
+            },
+            "object_list": {
+                "label": "GraphQL Objects List",
+                "order": 9,
+                "type": "list",
+            },
+            "headers": {
+                "label": "Headers",
+                "order": 10,
+                "type": "str",
+                "required": False,
+            },
+        }
+
+    def is_query(self, graphql_query):
+        ast = parse(graphql_query)  # pyright: ignore
```

Comment:
This checks only first operation right? could we have e.g. 
```
query GetUserInfo {
  user(id: "1") {
    id
    name
    email
  }
}

mutation UpdateUserEmail {
  updateUser(id: "1", email: "newemail@example.com") {
    id
    email
  }
}
```
That would bypass the check and execute a mutation against graphql backend?

Reply:
Maybe sth like?
```
def is_query_only(self, graphql_query):
    try:
        ast = parse(graphql_query)
        for definition in ast.definitions:
            # Check if the operation type is 'query'; skip non-operation definitions like fragments
            if hasattr(definition, 'operation') and definition.operation != 'query':
                return False
        return True
    except GraphQLSyntaxError as e:
        # Handle syntax errors in the GraphQL query
        self._logger.error(f"Failed to parse GraphQL query. Error: {e}")
        return False
 ```

Reply:
Oh yes! updated as per suggestion